### PR TITLE
Mock etcd DB and scheduler tests use case example

### PR DESCRIPTION
**Pull request by mandoonandy**
Unit tests have become terribly complicated requiring too much system knowledge. This is the start of making unit tests simple to write and trying to return them to something closer to unit tests (or maybe functional tests since mocking every call is too painful).

Only the Instance object is currently available. The scheduler tests have been modified to use the mock_etcd class, reducing the 811 line file to 348 lines.

Comments on the approach?

### CODE REVIEW

File: shakenfist/etcd.py [hunk 1]
```diff
@@ -350,7 +346,7 @@ def put(objecttype, subtype, name, data, ttl=None):
 
 
 @retry_etcd_forever
-def create(objecttype, subtype, name, data, ttle=None):
+def create(objecttype, subtype, name, data, ttl=None):
```
Comment (mikalstill) [inline shakenfist/etcd.py:349]: LOL. Nice catch.

**Review by mikalstill — COMMENTED** [review]
I think this looks like a nice approach, I haven't liked the current unit tests for a while which is why I've ended up preferring to write CI tests than unit tests. What do you see yourself needing to do to turn this into something ready for review?

**Comment by mandoonandy** [comment]
Actually that's ready to land. We can add more comments and other objects over time. I would expect the default nodes might also be tweaked.

**Comment by mikalstill** [comment]
CI failures are environmental.

**Pull request merged**